rust-driver-makefile.toml

```diff
@@ -1,66 +1,424 @@
+dependencies = ["copy-sys-to-package", "copy-inf-to-package"]
+command = "inf2cat"
+args = [
+  "/driver:${WDK_BUILD_OUTPUT_DIRECTORY}/${CARGO_MAKE_CRATE_FS_NAME}_package",
+  "/os:10_NI_X64,10_VB_X64",                                                   # TODO: this should be a parameter
+  "/uselocaltime",
+]
+
+[tasks.generate-certificate]
+private = true
+condition_script = [
+  '''
+#!@duckscript
+
+out = exec certmgr.exe -put -s WDRTestCertStore -c -n WDRLocalTestCert ${WDK_BUILD_OUTPUT_DIRECTORY}/WDRLocalTestCert.cer
+if eq ${out.code} 0
+    echo WDRLocalTestCert found in WDRTestCertStore. Skipping certificate generation.
+    exit 1
+else
+    echo WDRLocalTestCert not found in WDRTestCertStore. Generating new certificate.
+    exit 0
+end
+''',
+]
+command = "makecert"
+args = [
+  "-r",
+  "-pe",
+  "-a",
+  "SHA256",
+  "-eku",
+  "1.3.6.1.5.5.7.3.3",
+  "-ss",
+  "WDRTestCertStore",                                   # TODO: this should be a parameter
+  "-n",
+  "CN=WDRLocalTestCert",                                # TODO: this should be a parameter
+  "${WDK_BUILD_OUTPUT_DIRECTORY}/WDRLocalTestCert.cer",
+]
+
+[tasks.copy-certificate-to-package]
+private = true
+dependencies = ["generate-certificate"]
+script_runner = "@rust"
+script_runner_args = [
+  "--base-path",
+  "${CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY}",
+]
 script = '''
-call "%VC_BUILD_DIR%"
-if not exist DriverCertificate.cer (
-  makecert -r -pe -ss PrivateCertStore -n CN=DriverCertificate DriverCertificate.cer
-) else (
-  echo Certificate already exists.
-)
-signtool sign /a /v /s PrivateCertStore /n DriverCertificate /fd certHash /t http://timestamp.digicert.com "%OUTPUT_DIR%\package\%CARGO_MAKE_CRATE_FS_NAME%.cat"
+//! ```cargo
+//! [dependencies]
+//! wdk-build = { path = "./crates/wdk-build", version = "0.1.0" }
+//! ```
+#![allow(unused_doc_comments)]
+
+wdk_build::cargo_make::copy_to_driver_package_folder(
+    wdk_build::cargo_make::get_wdk_build_output_directory().join("WDRLocalTestCert.cer"),
+);
 '''
 
+[tasks.signtool]
+private = true
+dependencies = ["inf2cat", "generate-certificate"]
+command = "signtool"
+args = [
+  "sign",
+  "/v",
+  "/s",
+  "WDRTestCertStore",                                                                                  # TODO: this should be a parameter
+  "/n",
+  "WDRLocalTestCert",                                                                                  # TODO: this should be a parameter
+  "/t",
+  "http://timestamp.digicert.com",
+  "/fd",
+  "SHA256",
+  "${WDK_BUILD_OUTPUT_DIRECTORY}/${CARGO_MAKE_CRATE_FS_NAME}_package/${CARGO_MAKE_CRATE_FS_NAME}.cat",
+]
+
+[tasks.package-driver]
+private = true
+dependencies = [
+  "copy-sys-to-package",
+  "copy-pdb-to-package",
+  "copy-inf-to-package",
+  "copy-map-to-package",
+  "copy-certificate-to-package",
+  "signtool",
+  "infverif",
+]
+
+[tasks.package-driver-flow]
+# Only run flow if the current package is marked as a driver
+condition_script = [
```

Comment:
With https://github.com/sagiegurari/cargo-make/issues/1025 being resolved in cargo make v0.37.8, these condition scripts could be refactored to be rust-based for better maintainability, but since its fully functional right now, I'll leave that as a future improvement.